ambry-frontend/src/main/java/com/github/ambry/frontend/UserQuotaRequestCostPolicy.java

```diff
@@ -112,7 +114,11 @@ private double calculateCapacityUnitCost(RestRequest restRequest, RestResponseCh
         double cost = Math.ceil(contentSize / CU_COST_UNIT);
         return (cost > 0) ? cost : 1;
       case GET:
-        if (restRequest.getUri().contains(Operations.GET_SIGNED_URL)) {
+        String uri = restRequest.getUri();
+        if (uri.contains(Operations.GET_SIGNED_URL)) {
+          return INDEX_ONLY_COST;
+        } else if (uri.contains(SubResource.BlobInfo.name()) || uri.contains(SubResource.UserMetadata.name())) {
```

Comment:
One extra thing I noticed is that any HEAD request should also fall into this condition regardless of if it has the BlobInfo subresource. This is because HEAD will only check blob metadata since it only needs to return response headers.

Reply:
Sorry ignore this, I didn't see that HEAD requests will fall through to the default block.